helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -123,8 +136,11 @@ public synchronized void refresh(HelixDataAccessor accessor) {
     }
 
     // Refresh resource controller specific property caches
+    refreshCustomizedStateConfig(accessor);
+    _customizedStateCache.refresh(accessor, _liveInstanceCache.getPropertyMap());
     refreshExternalViews(accessor);
```

Comment:
Maybe we shall consider moving these 3 calculations to the update stages in the async thread. But let's leave it here for now. We will revisit anyway.

Reply:
ok, will discuss in the controller PR.